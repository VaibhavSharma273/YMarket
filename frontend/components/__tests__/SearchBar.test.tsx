import * as React from 'react';
import renderer from 'react-test-renderer';

import SearchBar from '../SearchBar';

it(`renders correctly`, () => {
    const props = {
        clicked: jest.fn(),
        searchPhrase: 'hello',
        setSearchPhrease: jest.fn(),
        setClicked: jest.fn(),
    };
    const tree = renderer.create(<SearchBar {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});
