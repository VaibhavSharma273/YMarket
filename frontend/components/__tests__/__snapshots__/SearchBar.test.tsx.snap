// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 0,
        "paddingLeft": 10,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ededed",
          "borderRadius": 15,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "padding": 10,
          "width": "90%",
        }
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        onFocus={[Function]}
        placeholder="Search YMarket"
        placeholderTextColor="#adadad"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 20,
            "marginLeft": 10,
            "width": "83%",
          }
        }
        underlineColorAndroid="transparent"
        value="hello"
      />
      <Text />
    </View>
    <View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
